### PR TITLE
Add interactive `stack next`

### PR DESCRIPTION
**Pull request by Brookke**
out

## Proposal

A stack next command that supports interactively selecting the path you want to navigate through. 

For example if you have multiple stacks on top of your trunk (like below) and you run `av stack next`
image

It will now prompt you to ask which one you want to use:
image

This is a proposal and uses the github.com/erikgeiser/promptkit library, if this is accepted and you have a preferred prompt library let me know.

### CODE REVIEW

**Comment by draftcode** [comment]
Have you tried `av stack switch`? It provides an interactive branch selection.

**Comment by Brookke** [comment]
> Have you tried `av stack switch`? It provides an interactive branch selection.

@draftcode Thanks for the tip, switch is great. But, sometimes I find it easier to navigate relative from where I am with `next` and `prev`, however currently `next` feels very broken when a stack consists of forks. 

Is the plan to eventually deprecate next and prev in favour of switch?

Additionally, I would appreciate your thoughts on the suggestions in https://github.com/aviator-co/av/discussions/302 if you get a moment :)

**Comment by draftcode** [comment]
> > Have you tried `av stack switch`? It provides an interactive branch selection.
> 
> @draftcode Thanks for the tip, switch is great. But, sometimes I find it easier to navigate relative from where I am with `next` and `prev`, however currently `next` feels very broken when a stack consists of forks.
> 
> Is the plan to eventually deprecate next and prev in favour of switch?

OK. We don't have a plan to deprecate these commands. Anyway, can you follow the same pattern as `av stack switch` impl for using bubbletea? I think promptkit is based on bubbletea and can be used as a widget of that.

> Additionally, I would appreciate your thoughts on the suggestions in #302 if you get a moment :)

I'll take a look.

**Comment by Brookke** [comment]
> Anyway, can you follow the same pattern as `av stack switch` impl for using bubbletea? I think promptkit is based on bubbletea and can be used as a widget of that.

Will do this soon

**Comment by Brookke** [comment]
@draftcode I've refactored this to use the bubbletea model structure :)

File: cmd/av/stack_next.go [hunk 1]
```diff
@@ -93,3 +73,151 @@ func init() {
...
+
+	return branchCheckedOutMsg{}
+}
+
+type checkoutBranchMsg struct{}
+type nextBranchMsg struct{}
+type showSelectionMsg struct{}
+
+func (m stackNextModel) nextBranch() tea.Msg {
+	if m.lastInStack && len(m.currentBranchChildren()) == 0 {
```
[the remark this answers is not in the document]
Comment (Brookke) [reply cmd/av/stack_next.go]: Thanks for the spot! I've update this to only call it once.

**Pull request merged**